### PR TITLE
Feature/issue advi lp

### PR DESCRIPTION
**Pull request by dustinvtran**
#### Summary:
Renames `lp` to `lp__` in output of variational inference.

#### Intended Effect:
N/A

#### How to Verify:
Run models by building them, then running them, e.g. `make examples/bernoulli/bernoulli && cd examples/bernoulli && ./bernoulli variational data file=bernoulli.data.R`.

#### Side Effects:
N/A

#### Documentation:
Changed variational inference documentation to reflect that `lp__` is actually calculated now. See [stan pull request](https://github.com/stan-dev/stan/pull/1539)

#### Reviewer Suggestions:
N/A

### CODE REVIEW

**Comment by syclik** [comment]
Jenkins, ok to test.

**Comment by dustinvtran** [comment]
Just realized that the checks never passed for this pull request; I added the unit test to the `make/tests` file. Please re-review this and merge :dancer:

File: src/test/interface/variational_output_test.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+TEST_F(CmdStan, variational_default) {
+  run_command_output out = run_command(base_command + " variational");
+
+  ASSERT_EQ(0, out.err_code);
+
+  stan::mcmc::chains<> chains = parse_output_file();
+  ASSERT_EQ(1, chains.num_chains());
+  ASSERT_EQ(1001, chains.num_samples());
```
Comment (syclik) [inline src/test/interface/variational_output_test.cpp]: is this all we want to test?

**Comment by syclik** [comment]
Sorry -- I missed this from 3 weeks ago. Is that all you want to test? Just make sure it works? Or did you want to make sure it's non-zero (to prevent a bug of someone accidentally changing the underlying implementation)? Or something else?

**Comment by dustinvtran** [comment]
I think it makes sense to only check if the output exists. `lp__` is temporary and only exists in the output because it needs to be compatible for `write_iteration_csv()`. Ideally, we plan on storing other things/replacing things, and we'll need to make a unit test at that time after we've all decided on the output interface for ADVI.

**Pull request merged**